src/test/java/com/epam/tc/hw6/utilsHw6/AssertStep.java

```diff
@@ -0,0 +1,114 @@
+    }
+
+    @Step("Assert 4 images on the index page")
+    public void assertImages() {
+        indexPage = new IndexPage(driver);
+        for (WebElement i : indexPage.getImagesDisplayed()) {
+            softAssert.assertTrue(i.isDisplayed());
+        }
+        softAssert.assertAll();
+    }
+
+    @Step("Assert 4 texts on the Index Page under icons")
+    public void assertTextUnderImages() {
+        indexPage = new IndexPage(driver);
+        assertThat(indexPage.getTextUnderImages()).as("Wrong actual text under icons").isEqualTo(expectedText);
+    }
+
+    @Step("Assert iframe with “Frame Button” exist")
+    public void assertIframe() {
+        indexPage = new IndexPage(driver);
+        softAssert.assertTrue(indexPage.getFrameElement().isDisplayed());
+        softAssert.assertAll();
+    }
+
+    @Step("Frame Button exists, Switch to the iframe and check that there is “Frame Button")
+    public void assertButtonInFrame() {
+        indexPage = new IndexPage(driver);
+        driver.switchTo().frame(I_FRAME_ELEMENT);
+        assertThat(indexPage.getFrameButton().isDisplayed());
+    }
+
+    @Step("5 items in the left section")
+    public void assertLeftItemsText() {
+        leftSection = new LeftSection(driver);
+        assertThat(leftSection.getNavigationMenuText()).as("Incorrect items in the left section").isEqualTo(names);
+    }
+
+    @Step("Different elements page is opened with proper title")
+    public void assertDiffElementsTitle() {
+        assertThat(driver.getTitle()).as("Incorrect diff elements title is displayed").isEqualTo(DIFF_ELEMENTS_TITLE);
+    }
+
+    @Step("Clicking checkboxes and assert checkboxes are ticked")
+    public void assertCheckboxes() {
+        diffElementsPage = new DiffElementsPage(driver);
+        assertThat(diffElementsPage.selectWaterCheckbox().isSelected());
+        assertThat(diffElementsPage.selectWindCheckbox().isSelected());
+    }
+
+    @Step("Selecting selen radiobutton")
+    public void assertRadioButton() {
+        diffElementsPage = new DiffElementsPage(driver);
+        softAssert.assertTrue(diffElementsPage.selectSelenRadioButton().isSelected());
+        softAssert.assertAll();
+    }
+
+    @Step("Select yellow dropdown")
+    public void assertDropdown() {
+        diffElementsPage = new DiffElementsPage(driver);
+        softAssert.assertTrue(diffElementsPage.selectYellowDropdown().isSelected());
+        softAssert.assertAll();
+    }
+
+    @Step("Check logpanel text")
+    public void assertLogpanel() {
+        diffElementsPage = new DiffElementsPage(driver);
+        assertThat(diffElementsPage.getWaterCheckboxLog().isDisplayed());
+        assertThat(diffElementsPage.getWindCheckboxLog().isDisplayed());
+        assertThat(diffElementsPage.getMetalRadiobuttonLog().isDisplayed());
+        assertThat(diffElementsPage.getColorsDropdownLog().isDisplayed());
```

Comment:
Let's suppose customer decided to add 10 more elements as the result 10 more logs you need to check. In the current solution you need to add 10 more lines of code. However, if you use one basic locator for log lines in common, you can just substitute it with an appropriate element name by passing this name as a parameter and moreover you need only one page method for all the logs like isLogDisplayed(String/Enum elementName).

Reply:
Changed the logic of the test. I'm not sure that is exactly what you described, but any way now the logic of the test is more common. So now if customer wants to add some new elements, I will just add that element name in List<String> selectedElementsInLogPanel

And also I didn't remove other created webelement locators (with water other elements) because all previous homework are connected with this one and removing of that elements causes the errors in code.